### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by skozubenko1**
Part 1 & 2
Test cases for Actors

### CODE REVIEW

File: Inheritance/src/Inheritance/ActorExtension.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Inheritance
+{
+    public static class ActorExtension
+    {     
+        public static string Speak(this Actor actor, bool WomenArePresent = false)
+        {
+            if (actor == null)
```
Comment (kenny2892) [inline Inheritance/src/Inheritance/ActorExtension.cs]: You should use "is null" instead of "== null" just in-case the equal operator has been overwritten.

Comment (skozubenko1) [reply Inheritance/src/Inheritance/ActorExtension.cs]: thank you for the suggestion, code corrected!

File: Inheritance/src/Inheritance/ActorExtension.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Inheritance
+{
+    public static class ActorExtension
+    {     
+        public static string Speak(this Actor actor, bool WomenArePresent = false)
```
Comment (breaman) [inline Inheritance/src/Inheritance/ActorExtension.cs]: parameters should be camelCase and womenArePresent should really be a property on Raj, not a parameter into this method.

File: Inheritance/src/Inheritance/ActorExtension.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                throw new ArgumentNullException();
+
+            switch (actor)
+            {
+                case Penny p:
+                    return p.Speaking();
+                case Sheldon s:
+                    return s.Speaking();                    
+                case Raj r:
+                    if (WomenArePresent)
```
Comment (breaman) [inline Inheritance/src/Inheritance/ActorExtension.cs]: this check should have been done on the case statement, not as a separate expression.

File: Inheritance/src/Inheritance/Penny.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Inheritance
+{
+    public class Penny : Actor
+    {
+        public string Speaking() => "Pengny is Speaking";
```
Comment (breaman) [inline Inheritance/src/Inheritance/Penny.cs]: my guess is Pengny is a typo, be careful of that.

File: Inheritance/test/Inheritance.Tests/ActorTests.cs [hunk 5]
```diff
@@ -5,9 +5,32 @@ namespace Inheritance.Tests
...
+            var talk = actor.Speak(WomenArePresent);
+            Assert.IsNotNull(talk);
+            Assert.IsTrue(talk.Length > 0, $"{actor.GetType()} didn't say anything");
+            return talk;
+        }
+
+        [TestMethod]
+        public void TestMethodPenny()
+        {
+            TestActor(new Penny());
```
Comment (breaman) [inline Inheritance/test/Inheritance.Tests/ActorTests.cs]: The test wrapper is nice to show that something is said, but you aren't verifying that what was said is what was expected.

File: Inheritance/test/Inheritance.Tests/PrinterTests.cs [hunk 6]
```diff
@@ -34,9 +35,40 @@ public void PersonGetsPrinted()
                 }
             }
         }
+        [TestMethod]
```
Comment (breaman) [inline Inheritance/test/Inheritance.Tests/PrinterTests.cs]: Where is the test for the Food item?

**Pull request closed without merging**